addons/pos_self_order/models/product_product.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from typing import List, Dict, Optional
+
+from odoo import models
+
+from odoo.addons.point_of_sale.models.pos_config import PosConfig
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_name(self) -> str:
+        """
+        Returns the name of the product without the code.
+        ex: product_sudo.display_name is '[FURN_7888] Desk Stand with Screen (Red)'
+        :return: 'Desk Stand with Screen (Red)' (we remove the [FURN_7888] part)
+        """
+        self.ensure_one()
+        # product_sudo.code would be 'FURN_7888', but we also need to remove the brackets
+        # and the space after the brackets (hence the +3)
+        return self.code and self.display_name[len(self.code) + 3 :] or self.display_name
```

Comment:
maybe this is better: `product_id.with_context(display_default_code=False)`

Reply:
much better indeed! thanks